arch/risc-v/src/litex/litex_head.S

```diff
@@ -188,17 +188,17 @@ exception_common:
  *  Name: g_intstackalloc and g_intstackbase
  ************************************************************************************/
 
-#if CONFIG_ARCH_INTERRUPTSTACK > 3
+#if CONFIG_ARCH_INTERRUPTSTACK > 15
   .bss
-  .align  4
+  .balign  16
   .global g_intstackalloc
   .global g_intstackbase
   .type   g_intstackalloc, object
   .type   g_intstackbase, object
 g_intstackalloc:
-  .skip  ((CONFIG_ARCH_INTERRUPTSTACK & ~3))
+  .skip  ((CONFIG_ARCH_INTERRUPTSTACK & ~15))
 g_intstackbase:
   .skip  4
   .size  g_intstackbase, 4
-  .size  g_intstackalloc, (CONFIG_ARCH_INTERRUPTSTACK & ~3)
+  .size  g_intstackalloc, (CONFIG_ARCH_INTERRUPTSTACK + 8) & ~15)
```

Comment:
Actually, you applied the `+ 8` to the wrong place.
It should have beed added to line 199, to the `.skip` command